src/storage/cacheInvalidation.ts

```diff
@@ -0,0 +1,19 @@
+import { removeItem } from './localStorage';
+import { keyString, StorageKey } from './storageKey';
+
+export function clearInvalidVersions(currentKey: StorageKey): void {
+  const { key, version } = currentKey;
+
+  // special case, if the cached version precedes versioning, clear it
+  removeItem(key);
```

Comment:
Just to clarify:

Let's say current version = 2, and you have three keys in local storage: 
1. `foo`
2. `foo.1`
3. `foo.2`

This line handles (1), and the for loop handles (2), right?

Reply:
Yes